### DISCUSSION TITLE
Custom Linux kernel module

### DISCUSSION THREAD

**Discussion by tysonite** [community/community#26614] [Actions]
<p>Hello,</p>
<p>Is there any way to request addition of kernel module to the VMs running Ubuntu? Right now, there are only serveral modules that can be added via modprobe, but I would like to have more modules available out of the box, for example, <a href="https://cateee.net/lkddb/web-lkddb/CAN_VCAN.html" rel="nofollow noopener">vcan</a> module.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>I am sorry to tell you that github action doesn’t have plan to add additional kernel module to hosted Ubuntu runner.  There are current installed software lists in Ubuntu runner.</p>
<p><a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fsoftware-installed-on-github-hosted-runners%23ubuntu-1804-lts&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7Ccf1635b126d14d452f9108d7820a449d%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637120852920118517&amp;sdata=BDclzdxwLmJR0eBdWoCcE5WFqa5xvP%2BGcvLo70Z63A4%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#ubuntu-1804-lts</a></p>
<p>If you are using private repository, I would suggest you use self-hosted runner with vcan installed.</p>

**Comment by ldorau** [comment]
<p>Hi,</p>
<ol>
<li>Has anything changed since then (Dec’19) ?<br>
If no:</li>
<li>Is there any paid-plan that allows installing custom kernel modules (SoftRoCE) on the hosted linux runners?</li>
</ol>
